### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadInterruptedException
  id: ThreadInterruptedException
  children:
  - System.Threading.ThreadInterruptedException.#ctor
  - System.Threading.ThreadInterruptedException.#ctor(System.String)
  - System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException
  fullName: System.Threading.ThreadInterruptedException
  type: Class
  summary: "时，将引发的异常<xref href=&quot;System.Threading.Thread&quot;></xref>处于等待状态时中断。"
  remarks: "创建线程后，它是一个或多<xref:System.Threading.ThreadState>状态中，直到它被销毁。</xref:System.Threading.ThreadState> 调用<xref:System.Threading.Thread.Interrupt%2A>当线程处于<xref:System.Threading.ThreadState>状态将导致在目标线程中引发 ThreadInterruptedException。</xref:System.Threading.ThreadState> </xref:System.Threading.Thread.Interrupt%2A> 如果线程处于<xref:System.Threading.ThreadState>状态，直到该线程进入该状态，不会引发异常。</xref:System.Threading.ThreadState> 如果线程永远不会阻止，则它会顺利完成而被中断。       ThreadInterruptedException 使用 HRESULT COR_E_THREADINTERRUPTED，其值 0x80131519。       ThreadInterruptedException 的实例的初始属性值的列表，请参阅<xref:System.Threading.ThreadInterruptedException.%23ctor%2A>构造函数。</xref:System.Threading.ThreadInterruptedException.%23ctor%2A>"
  example:
  - "The following code example shows the behavior of a running thread when it is interrupted and subsequently gets blocked.  \n  \n [!code-cpp[System.Threading.Thread.Interrupt#1](~/add/codesnippet/cpp/t-system.threading.threa_1_1.cpp)]\n [!code-vb[System.Threading.Thread.Interrupt#1](~/add/codesnippet/visualbasic/t-system.threading.threa_1_1.vb)]\n [!code-cs[System.Threading.Thread.Interrupt#1](~/add/codesnippet/csharp/t-system.threading.threa_1_1.cs)]"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadInterruptedException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>使用默认属性的类。"
  remarks: "下表显示<xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|本地化的错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException ();
    parameters: []
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>与指定的错误消息的类。"
  remarks: "下表显示<xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException (string message);
    parameters:
    - id: message
      type: System.String
      description: "解释异常原因的错误消息。"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>用序列化数据的类。"
  remarks: "在反序列化，以便重建通过流传输的异常对象过程中调用此构造函数。 有关详细信息，请参阅[XML 和 SOAP 序列化](~/add/includes/ajax-current-ext-md.md)。"
  syntax:
    content: protected ThreadInterruptedException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "<xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref>它保存关于所引发异常的序列化的对象数据。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "<xref:System.Runtime.Serialization.StreamingContext>它包含有关源或目标的上下文信息。</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>使用指定的错误消息和对导致此异常的内部异常的引用的类。"
  remarks: "前一个异常的直接结果应包含中的前一个异常的引用引发的异常，<xref:System.Exception.InnerException%2A>属性。</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>属性返回相同的值传递到构造函数中，或`null`如果<xref:System.Exception.InnerException%2A>属性不会提供给构造函数的内部异常值。</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       下表显示<xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|内部异常引用。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "解释异常原因的错误消息。"
    - id: innerException
      type: System.Exception
      description: "导致当前异常的异常。 如果`innerException`参数不是<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>中, 引发当前异常<xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref>处理内部异常的块。"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadInterruptedException.#ctor
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadInterruptedException.#ctor*
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException.ThreadInterruptedException
